### DISCUSSION TITLE
Anyone used an Orange Pi Zero or any other Raspberry Pi Zero alternative?

### DISCUSSION THREAD

**Discussion by chrisdaun** [aqualinkd/AqualinkD#263] [Q&A] [answered]
Curious if anyone has used one of these [Orange Pi Zero 2W](https://www.amazon.com/dp/B0CHM8W9YG/ref=twister_B0CHM7VWKV?_encoding=UTF8&psc=1) boards for this project and had success.

Thanks!

**Comment by sfeakes** [comment, marked as answer]
It should run perfectly fine on an Orange Pi Zero.   I know others run it on orange hardware.  Depending on OS used, you may need to compile it yourself.

**Reply by chrisdaun** [reply]
Thanks @sfeakes 
I did get it up and running on an Orange pi zero 2w running Ubuntu server. I used a sub $1 uart-to-rs485 board instead of the usb serial adapter and a buck converter to power off the bus like you have documented. Also setup the HomeKit integration with homebridge.

Amazing project, thanks for all the hard work!

**Reply by chrisdaun** [reply]
![image](https://github.com/sfeakes/AqualinkD/assets/1088495/4342f225-67e1-434f-9206-acb0c8b0ca28)

**Reply by chrisdaun** [reply]
![image](https://github.com/sfeakes/AqualinkD/assets/1088495/acc6be18-4997-448b-9aed-79055e9d104b)

**Reply by sfeakes** [reply]
Very glad you got it all running.
Do you have a link to the uart-to-rs485 you have used.

**Reply by chrisdaun** [reply]
@sfeakes 

uart-to-rs485 board
https://a.aliexpress.com/_m0emtwI

buck converter board
https://a.aliexpress.com/_mqAnDL0

Tip for anyone looking to go this route:
The adapter board can run at 3.3v or 5v but 3.3v wasn't enough power to send messages back to the Aqualink bus. I needed to give it 5v then everything worked fine. I'm powering the rs485 board off the orange pi's 5v.

Maybe I'll write up a quick guide on my setup if others want to follow.

**Reply by sjames99** [reply]
@chrisdaun I would be interested in a quick guide.  I had been looking at some other options as my last aqualinkd setup didn't make it and since you already had it working no reason to try and reverse it.

**Reply by chrisdaun** [reply]
@sjames99 I'll try and get the high level bits together when I have some time next week and then I can add more detail if people need it.
Stay tuned :)

**Reply by sjames99** [reply]
@chrisdaun just thought I would circle back and see if you had a chance to document your steps.

**Reply by chrisdaun** [reply]
> @chrisdaun just thought I would circle back and see if you had a chance to document your steps.

@sjames99 I did re-deploy my setup and I did capture some notes along the way. Happy to share what I have and add more detail where it's light or missing 😁

**Reply by sjames99** [reply]
That would be great!

**Comment by chrisdaun** [comment]
> That would be great!

Here you go
https://docs.google.com/document/d/1oaTGzKCac2eOb3nlkELobyPfplSXrlZATgusxWKnMqk
Comments enabled :)
